src/Proto3/Wire/Decode.hs

```diff
@@ -114,16 +118,17 @@ zigZagDecode i = shiftR i 1 `xor` (-(i .&. 1))
 -- We don't know what's inside some of these fields until we know what type
 -- we're deserializing to, so we leave them as 'ByteString' until a later step
 -- in the process.
-data ParsedField = VarintField Word64
-                 | Fixed32Field B.ByteString
-                 | Fixed64Field B.ByteString
-                 | LengthDelimitedField B.ByteString
-    deriving (Show, Eq)
+data ParsedField
+  = VarintField Word64
+  | Fixed32Field B.ByteString
+  | Fixed64Field B.ByteString
+  | LengthDelimitedField B.ByteString
+  deriving (Eq, Show)
 
 -- | Convert key-value pairs to a map of keys to a sequence of values with that
 -- key, in their reverse occurrence order.
 --
--- >>> toMap ([(FieldNumber 1, 3),(FieldNumber 2, 4),(FieldNumber 1, 6)] :: [(FieldNumber,Int)])
+-- >>> toMap ([(1,3),(2,4),(1,6)] :: [(FieldNumber,Int)])
```

Comment:
This is documentation, so it's probably better to leave in the explicit newtype constructor.